elytron/src/main/java/org/wildfly/extension/elytron/DomainDefinition.java

```diff
@@ -550,5 +557,155 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
         }
     }
 
+    static class ReadSecurityDomainIdentityHandler extends ElytronRuntimeOnlyHandler {
+
+        public static final SimpleAttributeDefinition NAME = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.NAME, ModelType.STRING, false)
+                .setAllowExpression(false)
+                .build();
+
+        static void register(ManagementResourceRegistration resourceRegistration, ResourceDescriptionResolver descriptionResolver) {
+            resourceRegistration.registerOperationHandler(new SimpleOperationDefinition(ElytronDescriptionConstants.READ_IDENTITY, descriptionResolver, NAME), new ReadSecurityDomainIdentityHandler());
+        }
+
+        private ReadSecurityDomainIdentityHandler() {
+        }
+
+        @Override
+        protected void executeRuntimeStep(final OperationContext context, final ModelNode operation) throws OperationFailedException {
+            // Retrieving the modifiable registry here as the SecurityDomain is used to create a new authentication
+            // context.
+            ServiceRegistry serviceRegistry = context.getServiceRegistry(true);
+            RuntimeCapability<Void> runtimeCapability = SECURITY_DOMAIN_RUNTIME_CAPABILITY.fromBaseCapability(context.getCurrentAddressValue());
+            ServiceName domainServiceName = runtimeCapability.getCapabilityServiceName(SecurityDomain.class);
+            ServiceController<SecurityDomain> serviceController = getRequiredService(serviceRegistry, domainServiceName, SecurityDomain.class);
+            SecurityDomain domain = serviceController.getValue();
+            ServerAuthenticationContext authenticationContext = domain.createNewAuthenticationContext();
+            String principalName = NAME.resolveModelAttribute(context, operation).asString();
+
+            try {
+                authenticationContext.setAuthenticationName(principalName);
+
+                if (!authenticationContext.exists()) {
+                    context.getFailureDescription().add(ROOT_LOGGER.identityNotFound(principalName));
```

Comment:
This applies generally. Why is 'add' being used instead of 'set'? The failure description should be ModelType.STRING not ModelType.LIST unless there's a real good reason it needs to be a list.